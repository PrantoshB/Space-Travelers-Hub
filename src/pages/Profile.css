.profile-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 1rem;
}

.profile-tbody .profile-tr {
  background: white !important;
  height: 3rem;
  font-size: 1.2rem;
  display: flex;
  align-items: center;
  padding: 0 4%;
  border-bottom: 2px solid #e7e7e7;
}

.profile-tbody .profile-tr:nth-last-child(1) {
  border: none;
}

.profile-td {
  border: none;
  padding: 0;
}
