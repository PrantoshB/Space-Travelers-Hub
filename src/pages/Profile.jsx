import React from 'react'

const Profile = () => {
  return (
    <h1>My Profile</h1>
  )
}

export default Profile