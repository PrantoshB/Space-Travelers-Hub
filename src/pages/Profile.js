import React, { useEffect } from 'react';
import './Profile.css';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rocket';
import { getMissionsAsync } from '../redux/missions/missions';

const Profile = () => {
  const dispatch = useDispatch();

  const data = useSelector((state) => state.rockets);
  const missionsData = useSelector((state) => state.missions);

  const { status, rockets } = data;
  const { missionsStatus, missions } = missionsData;

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchRockets());
      dispatch(getMissionsAsync());
    }
  }, [dispatch, status]);

  useEffect(() => {
    if (missionsStatus === 'idle') {
      dispatch(getMissionsAsync());
    }
  }, [dispatch, missionsStatus]);

  return (
    <div className="profile-container">
      <div className="mymissions">
        <h2>My Missions</h2>
        <table>
          <tbody className="profile-tbody">
            {
              missions.filter((mission) => (
                mission.isReserved === true
              )).map((mission) => (
                <tr className="profile-tr" key={mission.id}>
                  <td className="profile-td">{mission.name}</td>
                </tr>
              ))
            }
          </tbody>
        </table>
      </div>
      <div className="myrockets">
        <h2>My Rockets</h2>
        <table>
          <tbody className="profile-tbody">
            {
              rockets.filter((rocket) => (
                rocket.reserved === true
              )).map((rocket) => (
                <tr className="profile-tr" key={rocket.id}>
                  <td className="profile-td">{rocket.name}</td>
                </tr>
              ))
            }
          </tbody>
        </table>
      </div>
      <div className="mydragons">
        <h2>My Dragons</h2>
      </div>
    </div>
  );
};
export default Profile;
