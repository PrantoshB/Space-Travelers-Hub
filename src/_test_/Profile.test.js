import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../pages/Profile';
import store from '../redux/configureStore';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}><Profile /></Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
