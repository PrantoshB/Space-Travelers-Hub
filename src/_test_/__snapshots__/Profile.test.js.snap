// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="profile-container"
>
  <div
    className="mymissions"
  >
    <h2>
      My Missions
    </h2>
    <table>
      <tbody
        className="profile-tbody"
      />
    </table>
  </div>
  <div
    className="myrockets"
  >
    <h2>
      My Rockets
    </h2>
    <table>
      <tbody
        className="profile-tbody"
      />
    </table>
  </div>
  <div
    className="mydragons"
  >
    <h2>
      My Dragons
    </h2>
  </div>
</div>
`;
